src/libpmemfile/preload.c

```diff
@@ -1369,10 +1376,7 @@ lookup_pd_by_inode(struct stat *stat)
 	for (int i = 0; i < pool_count; ++i) {
 		struct pool_description *p = pools + i;
 		if (same_inode(&p->stat, stat)) {
-			PMEMfilepool *pfp;
-
-			pfp = __atomic_load_n(&p->pool, __ATOMIC_SEQ_CST);
-			if (pfp == NULL)
+			if (p->pool == NULL)
```

Comment:
Hm, arguably, this should still be an atomic read, and should make sure once a non-null value is read here, the values read after are the ones written to `*p` before `p->pool` was set to non-null.